src/paths/bellman_ford.c

```diff
@@ -203,3 +210,285 @@ int igraph_shortest_paths_bellman_ford(const igraph_t *graph,
 
     return 0;
 }
+
+
+/**
+ * \ingroup structural
+ * \function igraph_get_shortest_paths_bellman_ford
+ * \brief Calculates the weighted shortest paths from/to one vertex 
+ * allowing negative weights.
+ *
+ * If there are no negative ]weights, you are better off with 
+ * \ref igraph_shortest_paths_dijkstra() .
+ *
+ * \param graph The input graph, can be directed.
+ * \param vertices The result, the ids of the vertices along the paths.
+ *        This is a pointer vector, each element points to a vector
+ *        object. These should be initialized before passing them to
+ *        the function, which will properly clear and/or resize them
+ *        and fill the ids of the vertices along the geodesics from/to
+ *        the vertices. Supply a null pointer here if you don't need
+ *        these vectors. Normally, either this argument, or the \c
+ *        edges should be non-null, but no error or warning is given
+ *        if they are both null pointers.
+ * \param edges The result, the ids of the edges along the paths.
+ *        This is a pointer vector, each element points to a vector
+ *        object. These should be initialized before passing them to
+ *        the function, which will properly clear and/or resize them
+ *        and fill the ids of the vertices along the geodesics from/to
+ *        the vertices. Supply a null pointer here if you don't need
+ *        these vectors. Normally, either this argument, or the \c
+ *        vertices should be non-null, but no error or warning is given
+ *        if they are both null pointers.
+ * \param from The id of the vertex from/to which the geodesics are
+ *        calculated.
+ * \param to Vertex sequence with the ids of the vertices to/from which the
+ *        shortest paths will be calculated. A vertex might be given multiple
+ *        times.
+ * \param weights The edge weights. There mustn't be any closed loop in
+ *    the graph that has a negative total weight (since this would allow
+ *    us to decrease the weight of any path containing at least a single
+ *    vertex of this loop infinitely). If this is a null pointer, then the
```

Comment:
Let's document what the function will do if there is such a closed loop in the graph. (I guess it would just spin infinitely, but let's make it explicit).

Reply:
Scratch that, I see that we have an error code to indicate negative loops.